File: Java/Hardware.java
class Hardware extends Product{
    private String brand;
    private String model;

    Hardware(){
    }

    Hardware(String b, String m){
        this.brand = b;
        this.model = m;
    }

    void setBrand(String m){
        this.brand = m;
    }
    String getBrand(){
        return this.brand;
    }

    void setModel(String m){
        this.model = m;
    }
    String getModel(){
        return this.model;
    }
}

File: CPP/Hardware.cpp
#include <iostream>
#include <string>

class Hardware : public Product{
    private:
        string brand;
        string model;

    public:
        Hardware(){
        }

        Hardware(string b, string m){
            this->brand = b;
            this->model = m;
        }

        void setBrand(string b){
            this->brand = b;
        }
        string getBrand(){
            return this->brand;
        }

        void setModel(string m){
            this->model = m;
        }
        string getModel(){
            return this->model;
        }

        ~Hardware(){
        }
};